### PR TITLE
chore: optimize webpack config

### PR DESCRIPTION
**Pull request by webdiscus**
Hello @MarkVabulas,

I'm the author of the [html-bundler-webpack-plugin](https://github.com/webdiscus/html-bundler-webpack-plugin) 
that you already use in Webpack, but the configuration is not optimal and contains needles code.

I want help you to optimise and simplify the Webpack configuration.
You can build production bundle `index.html` containing embedded assets (script, style, images) into single HTML file by one step/stage.

The html-bundler-webpack-plugin replaces the functionality of many plugins and loaders such as:
  - html-webpack-plugin
  - html-inline-script-webpack-plugin
  - mini-css-extract-plugin
  - html-loader
  - style-loader

### Start the web app local in development mode
```
npm start
```

Open in browser the URL: `http://localhost:8080/` and you will see your sensor dashboard.
image

### Build production single HTML file
```
npm run build
```

In the `dist/` directory will be generated single `index.html` with inlined resources.

### P.S.

Two stages: compile and bundle are replaced with single "stage" - `build`.

You can try my [fork](https://github.com/webdiscus/demo-ultimon/tree/main).

### CODE REVIEW

File: Client/package.json [hunk 1]
```diff
@@ -18,25 +18,22 @@
...
-    "file-loader": "^6.2.0",
     "html-bundler-webpack-plugin": "^3.17.1",
-    "html-inline-script-webpack-plugin": "^3.2.1",
-    "html-webpack-plugin": "^5.6.0",
     "jquery": "^3.7.1",
     "npm-watch": "^0.13.0",
     "webpack": "^5.94.0",
-    "webpack-cli": "^5.1.4"
+    "webpack-cli": "^5.1.4",
+    "webpack-dev-server": "5.1.0"
```
Comment (webdiscus) [inline Client/package.json:36]: allow run your web app local in dev mode

File: Client/src/index.html [hunk 2]
```diff
@@ -35,39 +35,40 @@
...
-			<img style="left:786px; top:146px;" src="GraphLargeBG.png">
-			<img style="left:711px; top:185px;" src="CPU.png">
-			<img style="left:782px; top:146px;" src="GPU&GPU_Glass.png">
-			<img style="left:692px; top:300px;" src="GraphWideBG.png">
-			<img style="left:690px; top:370px;" src="GraphLargeMainBG.png">
-			<img style="left:691px; top:412px;" src="GraphCircleBG.png">
-			<img style="left:688px; top:411px;" src="Glass_Circle_Outer.png">
-			<img style="left:861px; top:412px;" src="GraphCircleBG.png">
-			<img style="left:858px; top:411px;" src="Glass_Circle_Outer.png">
+			<img style="left:51px; top:130px;" src="@images/LargeRectangleBG.png">
```
Comment (webdiscus) [inline Client/src/index.html:42]: use the Webpack alias to source directory with images

File: Client/src/index.js [hunk 3]
```diff
@@ -2,15 +2,4 @@ import {} from './Client.js';
 import {} from './DrawArcGuageDefault.js';
 import {} from './DrawGraphDefault.js';
 
-import {} from "./style.css";
```
Comment (webdiscus) [inline Client/src/index.js:5]: you can define source SCSS/CSS file directly in HTML with `<link>` tag

File: Client/src/index.js [hunk 4]
```diff
@@ -2,15 +2,4 @@ import {} from './Client.js';
...
-
-const importAll = require =>
```
Comment (webdiscus) [inline Client/src/index.js:7]: The Bundler Plugin resolves all source images. You don't need to use this "hack" to import images.

File: Client/src/index.html [hunk 5]
```diff
@@ -293,17 +294,18 @@
...
+			<img style="left:55px; top:361px;" src="@images/GraphMediumFG.png">
+			<img style="left:55px; top:451px;" src="@images/GraphMediumFG.png">
+			<img style="left:55px; top:542px;" src="@images/GraphMediumFG.png">
+			<img style="left:332px; top:297px;" src="@images/GraphWideFG.png">
+			<img style="left:692px; top:297px;" src="@images/GraphWideFG.png">
+			<img style="left:716px; top:439px;" src="@images/Glass_Circle_Inner.png">
+			<img style="left:886px; top:439px;" src="@images/Glass_Circle_Inner.png">
 		</div>
 
+		<script src="./index.js" defer="defer"></script>
```
Comment (webdiscus) [inline Client/src/index.html:309]: you can define source JS file directly in HTML with `<script>` tag

**Comment by MarkVabulas** [comment]
Hello @webdiscus!  

I greatly appreciate your review and I have investigated these changes.  I would like to point out a few different observations of the puzzle for you, so that you can understand some of the choices that I've made in my previous design.

- First, I wanted to say that I intentionally am avoiding using the webpack-dev-server for my development.  The goal is to allow users to make changes to their own index.html in the /src directory without having to understand the intricacies of how webpack works.  By leaving most of the changes internal to the html file, and not having to start a webpack-dev-server, it means that the main server software can be the one-true-source of the results.  It's possibly not ideal to be having the test environment be identical to the production one, but it guarantees consistent results.  Furthermore, the actual layout of the page is of secondary concern, the page is complete useless without the server software anyway, which also serves as the http host.  By forcing the UMS server to be the http server, it guarantees that the bundler is working properly, since it only responds to requests for index.html, and not for images or other asset files.
- I appreciate the webpack aliases for the source directories.  That's really useful.  I will include those changes, immediately.
- I like being able to use the html tags (for declaring the .js and .css) instead of importing them through requires statements.  This means the end-user can have finer control over what is loading at any given moment.

Thank you so very much for your help with this.  The project is now drastically improved for the client side editing than it was before.

I ended up using a large quantity of your suggestions, cleaned up the packages, and confirmed/tested that it is all working again.  Please check out the HEAD of main, and compare your changes to that one.  Can you review it again and tell me if I managed to adequately follow your best practices: https://github.com/MarkVabulas/ultimon/commit/58fbf01cedcb7c0d5300cb2d2e1ce1a30480c777

**Comment by webdiscus** [comment]
@MarkVabulas Your [58fbf01](https://github.com/MarkVabulas/ultimon/commit/58fbf01cedcb7c0d5300cb2d2e1ce1a30480c777) commit contains all needed changes, therefore this PR can be closed.

**Pull request closed without merging**